Создание ветки для PhoneBookApp

AntohaDontLookAtMe opened the pull request.

AntohaDontLookAtMe committed:
Создание ветки для PhoneBookApp

Dezodemius commented on PhoneBook/AbonentClass/Abonent.csproj:
```diff
@@ -0,0 +1,9 @@
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Давай уберём ImplicitUsings

Dezodemius commented on PhoneBook/PhoneBookApp/PhoneBookApp.csproj:
```diff
@@ -0,0 +1,19 @@
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
```
Давай уберём ImplicitUsings

Dezodemius commented on PhoneBook/PhoneBookApp/PhoneBookApp.csproj:
```diff
@@ -0,0 +1,19 @@
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\AbonentClass\Abonent.csproj" />
```
Не сильно понятно для чего отдельные  проекты них. Я бы Abonent и Phonebook поместил бы в один проект

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+{
+    public sealed class Phonebook
+    {
+        private static readonly Lazy<Phonebook> _instance = new Lazy<Phonebook>(() => new Phonebook());
```
Круто, что Lazy

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+    {
+        private static readonly Lazy<Phonebook> _instance = new Lazy<Phonebook>(() => new Phonebook());
+        public static Phonebook Instance => _instance.Value;
+        private readonly string FilePath = "phonebook.json";
```
Можно в константы, но не принципиально

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+            LoadPhoneBook();
+        }
+
+        /// <summary>
```
Круто, что есть дока

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+        public void AddAbonent()
+        {
+            Console.Clear();
+            Console.WriteLine("-----СОЗДАНИЕ АБОНЕНТА-----");
```
А давай весь код про консоль уберём из этого класса, а оставим только в Program.cs

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+        /// Поиск абонента по имени
+        /// </summary>
+        /// <param name="abonentName">Передаваемое имя абонента</param>
+        public void GetAbonentByName(string abonentName)
```
Пусть метод возвращает Abonent

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+        /// Поиск абонента по номеру телефона
+        /// </summary>
+        /// <param name="abonentPhoneNumber">Передаваемый номер телефона абонента</param>
+        public void GetAbonentByPhoneNumber(string abonentPhoneNumber)
```
Пусть метод возвращает Abonent

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+        /// <summary>
+        /// Вывод всех абонентов
+        /// </summary>
+        public void GetAllAbonents()
```
Пусть метод возвращает коллекцию абонентов

Dezodemius commented on PhoneBook/PhonebookClass/Phonebook.cs:
```diff
@@ -0,0 +1,231 @@
+            if (File.Exists(FilePath))
+            {
+                var json = File.ReadAllText(FilePath);
+                Abonents = JsonConvert.DeserializeObject<List<Abonent>>(json) ?? new List<Abonent>();
```
Круто, что json

AntohaDontLookAtMe committed:
Исправление замечаний

1. Перемещение Phonebook и  Abonent в один проект
2. Методы GetAbonentByName и GetAbonentByPhoneNumber возвращают Abonent
3. Метод GetAllAbonents возвращает List<Abonent>

Dezodemius commented on PhoneBook/PhoneBookApp/Phonebook.cs:
```diff
@@ -0,0 +1,267 @@
+        /// <summary>
+        /// Добавление абонента в телефонную книгу
+        /// </summary>
+        /// <param name="abonentName">Принимамое имя абонента</param>
```
Опечатка в слове Принимамое

Dezodemius replied:
ниже тоже есть

AntohaDontLookAtMe committed:
Исправление орфографических ошибок
